auth/client/src/test/java/org/wildfly/security/auth/client/DefaultSSLContextFromFileWorksAndHasPrecedenceTestCase.java

```diff
@@ -0,0 +1,29 @@
+package org.wildfly.security.auth.client;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import javax.net.ssl.SSLContext;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+
+public class DefaultSSLContextFromFileWorksAndHasPrecedenceTestCase {
+    private static final String CONFIG_FILE = "file:./src/test/resources/org/wildfly/security/auth/client/test-wildfly-config-client-default-sslcontext.xml";
+
+    @Test
+    public void testDefaultSSLContextFromFileWorksAndHasPrecedence() {
+        Security.insertProviderAt(new ElytronClientDefaultSSLContextProvider(CONFIG_FILE), 1);
+        Assert.assertNotNull(Security.getProvider("ElytronClientDefaultSSLContextProvider"));
+        AuthenticationContext.empty().run(() -> {   // provider will not use current auth context but will use authentication context from the file passed to provider. File passed to provider has precedence
+            SSLContext cip = null;
+            try {
+                cip = SSLContext.getDefault();
+            } catch (NoSuchAlgorithmException e) {
+                Assert.fail();
+            }
+            Assert.assertNotNull(cip);
+            Assert.assertEquals(cip.getProvider().getName(), ElytronClientDefaultSSLContextProvider.class.getSimpleName());
```

Comment:
IIUC, in case this is the "has precedence" part - and the following one is the "works" part - then I'd switch the two lines' order so that it matches the test method name.

Reply:
@fabiobrz switched